### PR TITLE
refactor: add verify() function

### PR DESCRIPTION
**Pull request by mukulljangid**
## High Level Overview of Change
Adds `verify()` which can take the `transaction` and check specifications for a particular `type`.

### Context of Change
Having one generic verification function to verify all types of transactions.

### Type of Change

- [x] New feature (non-breaking change which adds functionality)
- [X] Refactor (non-breaking change that only restructures code)

## Test Plan
Added tests for all types of transactions and the `verify()` function validates it and gives out the right `error`.

### CODE REVIEW

File: src/models/transactions/verify.ts
Comment (mvadari) [inline src/models/transactions/verify.ts:25]: This method should be in `Transaction` and should also be exported at the `src/models/transactions` level

File: test/models/verify.ts
Comment (mvadari) [inline test/models/verify.ts:12]: Nest tests that are for the same transaction type in another `describe`

File: src/models/transactions/verify.ts
Comment (natenichols) [inline src/models/transactions/verify.ts:85]: I would personally use a [switch](https://www.tutorialsteacher.com/typescript/typescript-switch) here, but

File: test/models/verify.ts
Comment (natenichols) [reply test/models/verify.ts:12]: What you could do for the tests is to just add `verify` to each of the existing tests, and expect the same result as the specific `verify` functions.

File: src/models/transactions/verify.ts
Comment (mukulljangid) [reply src/models/transactions/verify.ts]: Used switch!

File: test/models/verify.ts
Comment (mukulljangid) [reply test/models/verify.ts]: Added verify to all!

File: src/models/transactions/verify.ts
Comment (mukulljangid) [reply src/models/transactions/verify.ts]: Done!

File: src/models/transactions/transaction.ts
Comment (mvadari) [inline src/models/transactions/transaction.ts:60]: I'd put this after the switch, because those errors are more specific.

File: src/models/transactions/transaction.ts
Comment (mvadari) [inline src/models/transactions/transaction.ts:141]: Add the tr
```suggestion
      throw new ValidationError(`Invalid TransactionType: ${tx.TransactionType}`)
```

File: src/models/transactions/transaction.ts
Comment (mvadari) [inline src/models/transactions/transaction.ts:59]: Maybe also add a check for `onlyHasFields` (a helper function in `models`) before this equality is checked, as it will also help to provide better error messages

File: test/models/paymentTransaction.ts
Comment (mvadari) [inline test/models/paymentTransaction.ts:22]: Why did this change?

File: test/models/accountDelete.ts
Comment (mvadari) [inline test/models/accountDelete.ts:47]: Each of the `assert.throws` needs a separate `assert.throws` for `verify[SpecificTransaciton]` vs `verify` - `assert.throws` only checks if an error is thrown somewhere in the block, not that each throws an error

**Review by mvadari — CHANGES_REQUESTED** [review]

File: src/models/transactions/transaction.ts
Comment (mvadari) [inline src/models/transactions/transaction.ts:35]: You should instead do `import _ from "lodash"` and then use `_.function` elsewhere. Also, don't use `isUndefined` - it's much easier just to use `== null`.

File: src/models/transactions/transaction.ts
Comment (mvadari) [inline src/models/transactions/transaction.ts:36]: Run the linter on your new code - this doesn't look like it's even autofixing right now.

File: src/models/transactions/transaction.ts
Comment (mvadari) [inline src/models/transactions/transaction.ts:71]: What is the objective of this?

File: test-compiled-for-web/integration-test.js
Comment (mvadari) [inline test-compiled-for-web/integration-test.js:1]: This folder should be deleted - it was renamed to `testCompiledForWeb`

File: test/models/depositPreauth.ts
Comment (mvadari) [inline test/models/depositPreauth.ts:32]: Why was this test removed?

**Review by mvadari — CHANGES_REQUESTED** [review]
A few more changes + questions

File: src/models/transactions/transaction.ts
Comment (mvadari) [inline src/models/transactions/transaction.ts:2]: This should disable it for the whole file, instead of re-enabling at the bottom

File: src/models/transactions/transaction.ts
Comment (mvadari) [inline src/models/transactions/transaction.ts:72]: This is mildly incorrect TSdocs formatting.

Comment (natenichols) [reply src/models/transactions/transaction.ts:72]: Linter will fix this

**Review by natenichols — APPROVED** [review]
My only question is about the test that was removed, and that shouldn't hold up approval imo.

File: src/models/transactions/transaction.ts
Comment (mvadari) [reply src/models/transactions/transaction.ts:72]: I know, I just want to resolve linter errors in new code now, so we don't have to deal with it later

File: test/models/depositPreauth.ts
Comment (mukulljangid) [reply test/models/depositPreauth.ts]: Added it back!

File: src/models/transactions/transaction.ts
Comment (mukulljangid) [reply src/models/transactions/transaction.ts]: This has now been changed and done correctly

**Review by natenichols — APPROVED** [review]

**Comment by natenichols** [comment]
This looks good in its current form, just resolve the conficts.

File: test/models/accountDelete.ts
Comment (mvadari) [inline test/models/accountDelete.ts:39]: these files need to be linted properly

File: test/models/depositPreauth.ts
Comment (mvadari) [reply test/models/depositPreauth.ts:32]: It's still commented out?

File: test/models/escrowCreate.ts
Comment (mvadari) [inline test/models/escrowCreate.ts:154]: This should also throw instead of throwing instead

File: test/models/offerCancel.ts
Comment (mvadari) [inline test/models/offerCancel.ts:17]: Why was this replaced?

File: test/models/payment.ts
Comment (mvadari) [inline test/models/payment.ts:21]: Why was this replaced?

File: test/models/payment.ts
Comment (mvadari) [inline test/models/payment.ts:168]: This should work correctly as previously, right?

File: test/models/paymentTransaction.ts
Comment (mvadari) [inline test/models/paymentTransaction.ts:1]: This is an old file and shouldn't exist.

File: test/models/setRegularKey.ts
Comment (mvadari) [inline test/models/setRegularKey.ts:33]: This line shouldn't be commented out

File: test/models/signerListSet.ts
Comment (mvadari) [inline test/models/signerListSet.ts:97]: Run this through `verifySignerListSet` as well

File: test/models/signerListSet.ts
Comment (mvadari) [inline test/models/signerListSet.ts:48]: this should be camelCase: 
```suggestion
    assert.doesNotThrow(() => verify(signerListSetTx))
```

**Review by mvadari — COMMENTED** [review]
looks pretty good, a few minor comments

File: test/models/offerCancel.ts
Comment (mvadari) [inline test/models/offerCancel.ts:26]: This was left as before because it's probably a linter error for a too-long line - these models don't actually care whether the signature is correct or not

File: test/models/payment.ts
Comment (mvadari) [inline test/models/payment.ts:33]: ditto

File: test/models/payment.ts
Comment (mvadari) [inline test/models/payment.ts:20]: Why were things deleted from this object?

**Review by mvadari — APPROVED** [review]
please address all the comments before merging

File: test/models/offerCancel.ts
Comment (mukulljangid) [reply test/models/offerCancel.ts:26]: enode/decode won't work. it decodes it back to the sign without the dots and it does not match the original tx!

**Pull request merged**
